sdk/daml-lf/api-type-signature/src/test/scala/lf/typesig/reader/SignatureReaderSpec.scala

```diff
@@ -184,201 +185,224 @@ class SignatureReaderSpec extends AnyWordSpec with Matchers with Inside {
+              Ref.Identifier(itpPid, TIf),
+              Ref.Identifier(itpPid, LibTIf),
+            )
+            inside(choice1) { case TheUselessChoice(_, tIf) =>
+              tIf should ===(origin1.qualifiedName)
+            }
+            inside(choice2) { case TheUselessChoice(_, tIf) =>
+              tIf should ===(origin2.qualifiedName)
+            }
+        }
 
-    "resolve choices internally" in {
-      foundUselessChoice(
-        itp.main.resolveChoicesAndIgnoreUnresolvedChoices(PartialFunction.empty).typeDecls get Foo
-      )
-    }
+      "resolve inherited choices" in {
+        foundUselessChoice(itpES.typeDecls get Ref.Identifier(itpPid, Foo))
+      }
 
-    "collect direct and resolved choices in one map" in {
-      foundResolvedChoices(itpES.typeDecls get Ref.Identifier(itpPid, Foo))
-        .transform((_, cs) => cs.keySet) should contain theSameElementsAs Map(
-        Archive -> Set(
-          Some(Ref.Identifier(itpPid, TIf)),
-          Some(Ref.Identifier(itpPid, LibTIf)),
-          None,
-        ),
-        Useless -> Set(Some(Ref.Identifier(itpPid, TIf)), Some(Ref.Identifier(itpPid, LibTIf))),
-        Bar -> Set(None),
-      )
-    }
+      "resolve choices internally" in {
+        foundUselessChoice(
+          itp.main.resolveChoicesAndIgnoreUnresolvedChoices(PartialFunction.empty).typeDecls get Foo
+        )
+      }
 
-    // TODO: https://github.com/digital-asset/daml/issues/18821
-    // Make SignatureReaderSpec handle LF 1.x and active the 3 fallowing test for LF 1.x
-    "have interfaces with retroImplements" ignore {
-      itp.main.interfaces.keySet should ===(Set(LibTIf, TIf, RetroIf))
-      itp.main.interfaces(RetroIf).retroImplements should ===(
-        Set(Ref.TypeConName(itp.main.packageId, Foo))
-      )
-    }
+      "collect direct and resolved choices in one map" in {
+        foundResolvedChoices(itpES.typeDecls get Ref.Identifier(itpPid, Foo))
+          .transform((_, cs) => cs.keySet) should contain theSameElementsAs Map(
+          Archive -> Set(
+            Some(Ref.Identifier(itpPid, TIf)),
+            Some(Ref.Identifier(itpPid, LibTIf)),
+            None,
+          ),
+          Useless -> Set(Some(Ref.Identifier(itpPid, TIf)), Some(Ref.Identifier(itpPid, LibTIf))),
+          Bar -> Set(None),
+        )
+      }
 
-    "resolve retro implements harmlessly when there are none" ignore {
-      PackageSignature.resolveRetroImplements((), itpWithoutRetroImplements.all)((_, _) =>
-        None
-      ) should ===((), itpWithoutRetroImplements.all)
-      itpESWithoutRetroImplements.resolveRetroImplements should ===(itpESWithoutRetroImplements)
-    }
+      // TODO: https://github.com/digital-asset/daml/issues/18821
+      // Make SignatureReaderSpec handle LF 1.x and active the 3 fallowing test for LF 1.x
+      "have interfaces with retroImplements" in {
+        assume(v == LanguageVersion.Major.V1)
```

Comment:
FYI, I've discovered when working on the canton codebase that although the bazel test runner seems to be cool with tests cancelled with "assume", it is considered a failure when run from sbt. `assume` throws https://www.scalatest.org/scaladoc/3.2.18/org/scalatest/exceptions/TestCanceledException.html which indeed is semantically different from "I purposefully ignored this test".

Instead I had to use some manually written custom `ignore` parameterized by the language version. Unfortunately I can't find it anymore so it's probably been deleted since.